### PR TITLE
Adding A Test Case For Login Page

### PR DESCRIPTION
**Pull request by ChihweiLHBird**

### CODE REVIEW

**Comment by guzman-raphael** [comment]
@ChihweiLHBird Thanks for the PR! Are you running this currently on your side? If so, how are you running it to verify it tests properly?

**Comment by ChihweiLHBird** [comment]
> @ChihweiLHBird Thanks for the PR! Are you running this currently on your side? If so, how are you running it to verify it tests properly?

Thank you! It can be verified by run `npm test`. We can also add it to the GitHub action as a test automation.

**Comment by guzman-raphael** [comment]
@ChihweiLHBird Awesome, thanks for sharing. Will give it a try locally on my side.

Regarding how we go about CI and adopting it into the code pipeline, we normally include a test database that is only accessible locally for the tests. Have a look [here](https://github.com/datajoint/pharus/blob/7614e94638f3263bebc31cf029103b69b0c07340/docker-compose-test.yaml#L10) for reference. Since the scaffolding is not yet in this code base, we can help you with this. Let me give your PR a test locally on my end and we can contribute with you by PR'ing to your fork to help get you going. I'm thinking of including your tests (along with future tests) into the `dev` docker environment which will watch for changes and automatically re-run.

**Comment by ChihweiLHBird** [comment]
> @ChihweiLHBird Thanks for the PR! Are you running this currently on your side? If so, how are you running it to verify it tests properly?



> @ChihweiLHBird Awesome, thanks for sharing. Will give it a try locally on my side.
> 
> Regarding how we go about CI and adopting it into the code pipeline, we normally include a test database that is only accessible locally for the tests. Have a look [here](https://github.com/datajoint/pharus/blob/7614e94638f3263bebc31cf029103b69b0c07340/docker-compose-test.yaml#L10) for reference. Since the scaffolding is not yet in this code base, we can help you with this. Let me give your PR a test locally on my end and we can contribute with you by PR'ing to your fork to help get you going. I'm thinking of including your tests (along with future tests) into the `dev` docker environment which will watch for changes and automatically re-run.

Thank you! Would you like me to develop further test cases for other pages of the ReactJS App? I think we would need an account for testing purpose.

**Comment by guzman-raphael** [comment]
@ChihweiLHBird That would be great. Yes we can, I'll PR to your branch soon to help set that up with you.

**Comment by guzman-raphael** [comment]
Hey @ChihweiLHBird. BTW, in case you are having any image rendering and/or SSL/TLS related issues lately, our cert recently expired. You just need to update the `nginx` image with one that has been renewed i.e. bump the version. Like [this](https://github.com/datajoint/datajoint-labbook/blob/bb829a9bffd20313a8a79b311dee4f7b3096ccea/docker-compose-dev.yaml#L38).

**Comment by ChihweiLHBird** [comment]
> Hey @ChihweiLHBird. BTW, in case you are having any image rendering and/or SSL/TLS related issues lately, our cert recently expired. You just need to update the `nginx` image with one that has been renewed i.e. bump the version. Like [this](https://github.com/datajoint/datajoint-labbook/blob/bb829a9bffd20313a8a79b311dee4f7b3096ccea/docker-compose-dev.yaml#L38).

Okay, let me take a look. I am sorry, I didn't receive the notification from GitHub about your PR. I just merged it.

**Comment by ChihweiLHBird** [comment]
Hi @guzman-raphael , for the local test database, I just need to connect the local db in the labbook, and write the test cases for the pages after logging in with the root account? I can see the password of the root account is `labbook`. Is `root` the root account's username?

**Comment by guzman-raphael** [comment]
@ChihweiLHBird Yes, see line 12 that was just merged in `docker-compose-dev.yaml`.

**Comment by ChihweiLHBird** [comment]
> @ChihweiLHBird Yes, see line 12 that was just merged in `docker-compose-dev.yaml`.

Thank you!

**Comment by guzman-raphael** [comment]
@ChihweiLHBird Saw you pushed a new commit. Were you ready for review or not yet? Usually we `convert to draft` while still working through things and mark it `ready for review` to indicate to others to begin the review.

**Comment by ChihweiLHBird** [comment]
> @ChihweiLHBird Saw you pushed a new commit. Were you ready for review or not yet? Usually we `convert to draft` while still working through things and mark it `ready for review` to indicate to others to begin the review.

Although the test case cannot verify the behavior of the Web App after login operation, I think this PR is okay to be reviewed because it can simulate something on the login page. What do you think? Maybe we can get it reviewed (and merged) first and then adding other test cases later?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,11 @@ Observes [Semantic Versioning](https://semver.org/spec/v2.0.0.html) standard and
 - Docstrings for all classes and functions (#112) PR #124
 - Added sphinx docs and typedoc along with the docker and docker-compose file to build them. See `Developer Documentation/General Notes` for details. (#126) PR #129
 - Smaller screen layout (#58) PR #127
+- Documentation on running Jest tests w/ Docker. PR #130
+- Local testing database for `dev` Docker environment. PR #130
```
Comment (guzman-raphael) [inline CHANGELOG.md:11]: @ChihweiLHBird Would you add to our `CHANGELOG.md` and item to indicate that a new test for `/login` was added?

File: README.md [hunk 2]
```diff
@@ -13,9 +13,11 @@ DataJoint LabBook is a graphical user interface built using React to facilitate
 
 If you have not done so already, please install the following dependencies:
 
+
```
Comment (guzman-raphael) [inline README.md:16]: @ChihweiLHBird There is extra whitespace here and after the merge conflicts it looks like the docs section on **Run Tests w/ Docker** was removed.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@ChihweiLHBird That is a fair point. Reviewed the changes and there is only a few minor things to tweak before ready for merging. Looks like the merge conflict resolution omitted a section and introduced some more whitespace, etc. Will PR to you specifically what I mean and should be merge-ready once you accept/merge on your side.

File: README.md (hunk 2 above)
Comment (ChihweiLHBird) [reply README.md]: Merged, thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (ChihweiLHBird) [reply CHANGELOG.md:12]: Added in 3ac534f5d4c99e49d629dd9d199476bb0c36f364

**Review by guzman-raphael — APPROVED** [review]

**Pull request merged**